components/script/dom/globalscope.rs

```diff
@@ -78,6 +79,10 @@ pub struct GlobalScope {
     crypto: MutNullableDom<Crypto>,
     next_worker_id: Cell<WorkerId>,
 
+    /// A place to keep the clones read into this realm rooted.
+    #[ignore_malloc_size_of = "Rc<T> is hard"]
+    structured_clone_holder: Rc<StructuredCloneHolder>,
```

Comment:
I'm actually more interested to know why this has to be an `Rc<StructuredCloneHolder>` here instead of just `DomRefCell<Vec<DomRoot<Blob>>>`. In the `read_callback` function under structuredclone.rs, you can get a reference to the global by simply calling `GlobalScope::from_context(cs)`, so passing in a raw Rc pointer seems unnecessary to me.

Reply:
My answer to 'why a`Rc`' would be: 'it was easy to get a raw pointer out of it and cast it to `*mut raw::c_void`'.  Would that be doable with a `DomRefCell`? (I see that a `RefCell` has a [`as_ptr`](https://doc.rust-lang.org/std/cell/struct.RefCell.html#method.as_ptr) method, and it doesn't seem to be available on the `DomRefCell` variant.).

Assuming we can get a raw pointer using `as_ptr`, how would we build it back on the read end from this pointer? I don't see the equivalent of `Rc::from_raw` on `RefCell`...

Any ideas?

Reply:
Going back to what I suggested -- you already have a `*mut JSContext`, so you can make use of that to get a `GlobalScope` out of it, and in your `read_callback` function, you can simply call methods on `GlobalScope` to enqueue the `DomRoot<Blob>` into its data holder; there's no need to pass in a raw Rc pointer just to get a reference to the data holder.

Reply:
Awesome, I'll give it a try!